Kernel/Process.cpp

```diff
@@ -1803,6 +1804,34 @@ int Process::sys$mkdir(const char* pathname, mode_t mode)
     return VFS::the().mkdir(StringView(pathname, pathname_length), mode & ~umask(), current_directory());
 }
 
+int Process::sys$realpath(const char* pathname, char* buffer, ssize_t size)
+{
+    if (size < 0)
+        return -EINVAL;
+    if (!validate_read_str(pathname))
+        return -EFAULT;
+
```

Comment:
You also need to validate a write to `buffer` for `size` bytes.

Reply:
Done.